lib/phoenix/code_reloader/group_leader_proxy.ex

```diff
@@ -0,0 +1,64 @@
+# Spawned by the Proxy when capture is started,
+# to capture stdout of the compilation process.
+# Restores the original group leader when terminated after compilation.
+# Output is forwarded to the parent proxy with the :stdout message.
+defmodule Phoenix.CodeReloader.GroupLeaderProxy do
```

Comment:
What if you make this a regular process started via `start_link` inside `proxy` so you can re-use the handle_info and put_chars logic?